cloud/sts.go

```diff
@@ -61,6 +64,15 @@ type AssumeRoleCredentials struct {
 	Expiration      time.Time
 }
 
+// assumeRoleCache holds AWSCredentials for assumed roles.
+var assumeRoleCache = cache.WithOtel(cache.NewTTLInMemory[AssumeRoleCredentials](), "aws-assume-role")
+
+// minAssumeRoleCacheLifetime is the minimum lifetime of an assumed role
+// when retrieved from the cache. This is set low because the cache is currently
+// used in s3 credentials, which only need valid credentials for the
+// beginning of the operation.
+const minAssumeRoleCacheLifetime = 2 * time.Minute
```

Comment:
This not being customizable is a choice, I can't imagine why we would cache AssumeRole credentials other than this s3 workflow- but if we want in the future we could add an option in the AssumeRoleOptions (like `Cache`) that overrides this value for a particular request

Reply:
Does this assumption about the credentials also apply to multipart bucket uploads? Those are used for uploading large files to S3.

Reply:
I'm doing research on this but my first guess is _no_. If that is the case, I think we'll need to have a custom credential provider that calls Evergreen to refresh the credentials as they expire to have an effective cache

Reply:
It looks [like yes](https://stackoverflow.com/questions/70384905/aws-s3-temporary-credentials-expiring-before-upload-finishes). Reading their docs, it checks that it doesn't do some sneaky thing that would maybe continuing accepting the same credentials which makes sense. I'll rewrite this